http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -741,16 +415,58 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
 
       app = dbApp.app
 
-      // Delete WSM resources associated with the app
-      // Ideally an app only consists of WSM resources, as more resources are moved into WSM
-      // everything between deleteAppWsmResources and appQuery.markAsDeleted should be removed
-      _ <- deleteAppWsmResources(app, params.workspaceId)
+      // WSM deletion order matters here. Delete WSM database resources first.
+      wsmDatabases <- appControlledResourceQuery
+        .getAllForAppByType(app.id.id, WsmResourceType.AzureDatabase)
+        .transaction
+      _ <- wsmDatabases.traverse { database =>
+        deleteWsmResource(workspaceId, app, database)
+      }
+
+      // Then delete namespace resources
+      wsmNamespaces <- appControlledResourceQuery
+        .getAllForAppByType(app.id.id, WsmResourceType.AzureKubernetesNamespace)
+        .transaction
+      deletedNamespace <- wsmNamespaces
```

Comment:
does any of the delete ops need to be synchronous? like do we need to wait for namespace to be deleted first before deleting identities etc?

Reply:
Yes, exactly -- the deletion order needs to be: database, namespace, identity. Each of the `deleteWsm*Resource` methods waits for WSM deletion before returning.